src/app/pages/property-details/confirmation-modal/confirmation-modal.component.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { RouterTestingModule } from '@angular/router/testing';
+import { of, throwError } from 'rxjs';
+import {
+  resetSpies,
+  propertyDetailSpy,
+  toastServiceSpy,
+  routerSpy
+} from 'src/app/helpers/spies';
+import { ActivatedRoute, Router } from '@angular/router';
+import { NgxSpinnerModule } from 'ngx-spinner';
+import { ToastrService } from 'ngx-toastr';
+import { ConfirmationModalComponent } from './confirmation-modal.component';
+import { PropertyDetailService } from 'src/app/services/property-detail/property-detail.service';
+import { AppModule } from 'src/app/app.module';
+
+describe('ConfirmationModalComponent', () => {
+  let component: ConfirmationModalComponent;
+  let fixture: ComponentFixture<ConfirmationModalComponent>;
+
+  let mockActivateRouteParam = {
+    paramMap: of({ get: () => 'hello' })
+  };
+
+  beforeAll(() => resetSpies([propertyDetailSpy]));
+  afterEach(() => resetSpies([propertyDetailSpy]));
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ConfirmationModalComponent],
+      imports: [
+        HttpClientTestingModule,
+        RouterTestingModule.withRoutes([
+          { path: '**', component: ConfirmationModalComponent }
+        ]),
+        AppModule,
+        NgxSpinnerModule
+      ],
+      providers: [
+        {
+          provide: PropertyDetailService,
+          useValue: propertyDetailSpy
+        },
+        {
+          provide: ActivatedRoute,
+          useValue: mockActivateRouteParam
+        },
+        {
+          provide: Router,
+          useValue: routerSpy
+        },
+        {
+          provide: ToastrService,
+          useValue: toastServiceSpy
+        }
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ConfirmationModalComponent);
+    component = fixture.componentInstance;
+    propertyDetailSpy.deletePropertyService.and.returnValue(
+      of({ message: 'successfully deleted' })
+    );
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+  it('should call deleteProperty method', () => {
+    const result = { message: 'successfully deleted' };
+    propertyDetailSpy.deletePropertyService.and.returnValue(
+      of({ message: 'successfully deleted' })
+    );
+    component.deleteProperty();
+    expect(component.message).toEqual(result.message);
+  });
+  it('should throw an error', () => {
+    const errorMessage = {
+      detail: 'You dont have a permission to perform such action'
+    };
+    propertyDetailSpy.deletePropertyService.and.returnValue(
+      throwError({
+        errors: {
+          detail: 'You dont have a permission to perform such action'
+        }
+      })
+    );
+    component.deleteProperty();
+
+    expect(toastServiceSpy.error).toHaveBeenCalledWith(errorMessage.detail);
+  });
+
+  it('should display the modal method', () => {
```

Comment:
Hey @MBenedicte, I believe we can do better we this test.we can add an expectation to it.
Maybe we could select it from the dom
Then make sure it's visible
Run the closeModal method
Then check its visbility in the expectaion